orbit_predictor/predictors/base.py

```diff
@@ -279,6 +281,76 @@ def get_next_pass(self, location, when_utc=None, max_elevation_gt=5,
         else:
             raise NotReachable('Propagation limit date exceeded')
 
+    def eclipses_since(self, when_utc, limit_date=None):
+        """
+        An iterator that yields all eclipses start and end times between
+        when_utc and limit_date.
+
+        The next eclipse with a end strictly after when_utc will be returned,
+        possibly the current eclipse.
+        The last eclipse returned starts before limit_date, but it can end
+        strictly after limit_date.
+        No circular orbits are not supported, and will raise NotImplementedError.
+        """
+        def _get_illumination(t):
+            my_start = start + dt.timedelta(seconds=t)
+            result = get_satellite_minus_penumbra_verticals(
+                self.get_only_position(my_start),
+                my_start
+            )
+            return result
+
+        orbital_period_s = 60 / self.mean_motion * 2 * np.pi
```

Comment:
We already have a `self.period` property:

```suggestion
        orbital_period_s = self.period * 60
```